### PR TITLE
docs: lint NodeJS snippets

### PR DESCRIPTION
**Pull request by aluzzardi**
This uses the same eslint config as the SDK itself. /cc @vikram-dagger @slumbering @kpenfound 

If we go ahead with this -- we should probably do the same for Python as well (/cc @helderco)

Go is already covered by the main engine:lint

**NOTE**: Apparently eslint makes the SDK chaining a one-liner

### CODE REVIEW

File: docs/current/sdk/nodejs/snippets/.eslintrc.cjs
Comment (slumbering) [inline docs/current/sdk/nodejs/snippets/.eslintrc.cjs:13]: The line is not necessary. `types.ts` will be removed

**Comment by helderco** [comment]
Can we move the snippets into the sdk directory (e.g., `sdk/python/docs/snippets`)? It would allow reusing the same config and commands. When the [examples were moved](https://github.com/dagger/dagger/pull/3769) we got the same benefit.

**Comment by vikram-dagger** [comment]
> > we should probably do the same for Python as well
> 
> Can we move the snippets into the sdk directory (e.g., `sdk/python/docs/snippets`)? It would allow reusing the same config and commands. 

We can definitely do this. The one thing about it that bothers me (and I'm not sure if this is valid) is that currently the docs and everything related to them is in a single place ie. the docs/ directory. If we move these code samples to outside the docs directory, we will be fragmenting the docs across two locations, and I wonder if that could that create an issue in future?

**Comment by gerhard** [comment]
> @vikram-dagger We can definitely do this. The one thing about it that bothers me (and I'm not sure if this is valid) is that currently the docs and everything related to them is in a single place ie. the docs/ directory. If we move these code samples to outside the docs directory, we will be fragmenting the docs across two locations, and I wonder if that could that create an issue in future?

Would soft symlinks help?

**Comment by slumbering** [comment]
@aluzzardi The PR LGTM. Any reason for the Draft status ?
Otherwise I'll merge it. Thanks

**Comment by aluzzardi** [comment]
> @aluzzardi The PR LGTM. Any reason for the Draft status ?

Just wanted to check with you the prettier config before removing the draft status.

Also it needs a rebase, and there's a merge commit that made it into this PR:

Screen Shot 2022-11-16 at 3 17 15 PM

I'll fix that and remove the draft status

**Comment by aluzzardi** [comment]
Please be careful with merging `main` into PRs, and double check before merging any PR that the commit list is clean.

This is the git log of this PR right now (my original commit is the last line). It makes looking at the history quite impossible:

```
* b1fd7f17 (HEAD -> docs-lint-node, fork/docs-lint-node) Fix linter
*   15e3ce30 Merge branch 'main' into docs-lint-node
|\  
| * e46cf71e build(deps): bump github.com/containerd/containerd from 1.6.9 to 1.6.10 (#3817)
| * 19954e62 build(deps): bump node-fetch from 3.2.10 to 3.3.0 in /sdk/nodejs (#3807)
| * 072a6c1b build(deps): bump github.com/moby/buildkit from 0.10.5 to 0.10.6 (#3795)
| * dfb4b8b4 NodeJS Provisioning Follow ups  (#3861)
| * 325ac37b docs(python): port private registry docs to python (#3760)
| * a503b464 docs: Node.js SDK getting started: ignore node_modules (#3863)
| * 10e89ac6 sdk: python: Pass objects instead of IDs (#3870)
| * adaf88cf docs: Add FAQ entry for container registry auth (#3876)
| * 783f9437 build(deps): Bump strawberry-graphql in /sdk/python (#3869)
| * 5619f43f docs: Add sidebar for Node.js SDK docs (#3873)
| * 74389d55 docs: Format Node SDK  instructions using tabs (#3875)
| *   59a34e5a Merge pull request #3825 from KGB33/sdk-python-improve-error-messages
| |\  
| | * 48c9b623 Remove temp files on error.
| | * 4215a0e4 sdk: python: Refactor provision errors
| | * c9c85fdb Prevented the attempted read of an None or unreadable stderr.
| | * 2e82584b Improved error message when dagger engine is cached but docker is not running.
| | * d69f088a Improved error messages when docker is not installed/running.
| | * b3fe0b70 Added DaggerException.
| * |   cc426de4 Merge pull request #3816 from helderco/python-test-3.11
| |\ \  
| | * | c883eb01 sdk: python: Test Python 3.11
| * | |   32e16359 Merge pull request #3815 from helderco/python-dependabot
| |\ \ \  
| | * | | 4fb1b396 chore: Add other SDK labels from dependabot
| | * | | 3acf5616 chore(pyhon): Add dependabot
| | |/ /  
| * | |   76d28eb6 Merge pull request #3814 from helderco/python-fix-windows-platform
| |\ \ \  
| | * | | a321f59b sdk: python: Fix windows platform detection
| | |/ /  
| * | | c0a66164 remove unused types.ts file (#3868)
| * | | 7eb24f64 build(deps-dev): Bump typescript from 4.8.4 to 4.9.3 in /sdk/nodejs (#3853)
| * | |   dcce1a77 Merge pull request #3866 from aluzzardi/sdk-node-enable-prettier
| |\ \ \  
| | * | | 44d1bce2 (fork/sdk-node-enable-prettier) Remove unessary semicolons
* | | | | 8f30d2ba docs: fix format on nodejs snippets
* | | | | b40e8b54 docs: lint nodejs code snippets
```

File: docs/current/sdk/nodejs/snippets/.eslintrc.cjs
Comment (aluzzardi) [reply docs/current/sdk/nodejs/snippets/.eslintrc.cjs]: Done

**Comment by aluzzardi** [comment]
Also copied `.prettierrc` here

**Comment by gerhard** [comment]
⤴️ In summary:
- ✅ Always rebase PR branches on top of `main` - keeps history clean
- ❌ Do not merge `main` into PR branches - very hard to tell what changed

---

Related from ad-hoc discussions:

- ✅ Default to **Squash and merge** PRs
- ⚠️ On rare occasions **Create a merge commit**
  - 2-3 commits
  - when one of them is likely to be reverted soon after
- ✅ If in doubt, **Squash and merge**

**Review by slumbering — APPROVED** [review]
LGTM

**Pull request merged**
